src/foam/box/sf/SFManager.js

```diff
@@ -204,6 +206,27 @@ foam.CLASS({
           }
         });
       `
+    },
+    {
+      name: 'fclone',
+      type: 'FObject',
+      documentation: 'Override default fclone, skip properties that could cause infite loop',
+      javaCode:`
+        try {
+          FObject ret = getClass().newInstance();
+          List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
+          for ( PropertyInfo prop : props ) {
+            if ( ! prop.isSet(this) ) continue;
+            // properties that cause infite looping.
+            if ( foam.util.SafetyUtil.equals("prorityQueue", prop.getName()) ) continue;
```

Comment:
For these properties, you could just add this property property:
      javaCloneProperty: 'set(dest, get(source));',